### ISSUE TITLE
After update compile time errors are no longer reported

### ISSUE THREAD

**Issue by Skarlso** [joefitzgerald/go-plus#393]
Hi Joe!

First off, awesome job on this plugin. Second, now that I updated, on save, compile time issues are no longer reported. For example consider this code:

```
import "fmt"

func main() {
    fmt.Printasdf("Something")
}

```

This does not raise any errors in Atom, and it did previously. 

I don't know how to debug atom for error logs, I'm happy to provide them to you.

Thanks!

**Comment by chris-r** [comment]
This affects me, too.

**Comment by mlitvin** [comment]
Installing gotype helps (I'm not if it is the same) but IMHO this is a degradation in usability, a user just installing or updating go-plus would get into this problem and the solution is not that clear.

**Comment by michaeltrobinson** [comment]
This is definitely a problem, using `gotype` is not a reasonable solution as it is end-of-life and requires the packages to be `go installed`ed beforehand.

**Comment by joefitzgerald** [comment]
Actually, `cmd/gotype` is not end of life; it will be maintained and may be updated to work with `loader`. In the interim, the temporary fix is to run `go install ./...` in your package periodically. I am working on `https://github.com/joefitzgerald/builder-go` to do this for you automatically on save.

**Comment by Skarlso** [comment]
@joefitzgerald With all do respect, I don't want to install yet another plugin. I don't understand why this functionality was removed, or if it was an accident, why can't it be added back? It was working without a problem.

**Comment by pzinovkin** [comment]
No offence, but in my opinion this release brought us some sort of configuration hell.
After spending some time investigating which plugin (I remember installing only one!) causes troubles and trying to mitigate it by fiddling with settings, I've uninstalled plugin and installed previous version with apm:
`apm install go-plus@3.5.4`

**Comment by joefitzgerald** [comment]
@pzinovkin Sorry to hear you've had issues. The [release notes for 4.0.0](https://github.com/joefitzgerald/go-plus/releases/tag/v4.0.0) clearly call out the evolution of the package from a monolithic one to a collection of single purpose packages.

This strategy is already paying dividends: https://atom.io/packages/go-debug was released and it makes use of `go-config`.

You're of course welcome to keep using the old version of the package; if you'd like to describe the issues you were having, perhaps I or someone else who uses the package can help you out and resolve your issues?

**Comment by Skarlso** [comment]
I'm sorry, but the reason I was using go-plus was NOT having to install a mirrad of plugins. That was it's main selling point. 

So now... What's the point of it?

**Comment by pzinovkin** [comment]
I understand motivation behind go-config and separation and ok with it. go-config and go-get clearly states its purpose and do exactly what expected. But release notes says nothing that go build is not called anymore and gometalinter doesn't cover it.

**Comment by joefitzgerald** [comment]
@pzinovkin I honestly thought `gotype` covered the scenarios I was using `go build` for, and I wanted to have the package function well for people using gb. It's clear `gotype` has some gaps in its coverage of that set of functionality. **I've acknowledged that and stated that I am building a package to add back that functionality.**

@Skarlso the packages were installed for you; go-plus will continue to be the only package you have to install to get the curated set of functionality that it includes. There is admittedly some additional burden to figure out which package is causing a particular problem, but people are free to post issues here, and they do - as you can we are discussing in this forum still.

I fully acknowledge that change is hard. Some people might wish that go-plus remains a single package with all the code included within the same package; but that was causing issues related to maintenance and reuse, which was preventing contributions and preventing others from creating their own single purpose packages (e.g. `go-debug` is a good example of this kind of package).

**Comment by Skarlso** [comment]
I'm all for separation of concern. So, go-build not working, is an issue then?

**Comment by joefitzgerald** [comment]
@Skarlso https://github.com/joefitzgerald/go-plus/issues/394#issuecomment-199373404

**Comment by Skarlso** [comment]
Oh thanks. I didn't see that one. Do you want this issue to remain? I can close it if you like. :-)

**Comment by joefitzgerald** [comment]
I'm fine leaving this one open.

**Comment by Skarlso** [comment]
Okay. Thanks!

**Comment by joefitzgerald** [comment]
This is fixed in `go-plus` `4.0.0` https://github.com/joefitzgerald/go-plus/releases/tag/v4.1.0.
